indico/modules/events/registration/models/registrations.py

```diff
@@ -330,10 +343,12 @@ def update_state(self, approved=None, paid=None):
                 self.state = RegistrationState.unpaid
             elif approved:
                 self.state = RegistrationState.complete
+            elif rejected:
+                self.state = RegistrationState.rejected
         elif self.state == RegistrationState.unpaid:
             if paid:
                 self.state = RegistrationState.complete
-            elif approved is False:
+            elif approved is False and moderation_required:
```

Comment:
should this check for moderation required? I think if we call this it makes sense to always go to pending even if moderation is not enabled